src/app/pages/properties/my-wishlist/my-wishlist.component.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { HttpClientModule } from '@angular/common/http';
+import { RouterTestingModule } from '@angular/router/testing';
+import { PropertiesWishlistService } from 'src/app/services/properties/properties-wishlist/properties-wishlist.service';
+import { propertiesWishlistSpy, toastServiceSpy, routerSpy } from 'src/app/helpers/spies';
+import { ToastrService } from 'ngx-toastr';
+import { Router } from '@angular/router';
+import { of, throwError } from 'rxjs';
+import { By } from '@angular/platform-browser';
+
+describe('MyWishlistComponent', () => {
+  let component: MyWishlistComponent;
+  let fixture: ComponentFixture<MyWishlistComponent>;
+  const mockWihslistProperties = {
+    "data": {
+      "property": [
+        {
+          "id": 7,
+          "price": 20000000.0,
+          "title": "Gemini Court",
+          "property_type": "B",
+          "description": "Awesome property",
+          "slug": "g-c"
+        },
+        {
+          "id": 8,
+          "price": 1000000.0,
+          "title": "A test property",
+          "property_type": "B",
+          "description": "This is a house I will buy once in bukavu",
+          "slug": "test-test"
+        }
+      ]
+    }
+  };
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ MyWishlistComponent ],
+      imports: [
+        HttpClientModule,
+        RouterTestingModule],
+      providers: [
+        {
+          provide: PropertiesWishlistService,
+          useValue: propertiesWishlistSpy
+        },
+        {
+          provide: ToastrService,
+          useValue: toastServiceSpy
+        },
+        { provide: Router, useValue: routerSpy }]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MyWishlistComponent);
+    component = fixture.componentInstance;
+    propertiesWishlistSpy.getMyWishlist.and.returnValue(of(mockWihslistProperties));
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+  it('should show a toast with error when it fails to load wishlist items on init', () => {
+    const mockedResponse = {
+      "errors": "An error occurs when retrieving your wishlist"
+    }
+    propertiesWishlistSpy.getMyWishlist.and.returnValue(throwError(mockedResponse));
+    component.ngOnInit();
+    expect(toastServiceSpy.error).toHaveBeenCalledWith(mockedResponse.errors);
+    
+  });
+  it('test if it displays all the components in my wishlist', ()=>{
+    const htmlelments = fixture.debugElement;
+    expect(htmlelments.query(By.css('.list-title')).nativeElement.textContent).toContain('My wished properties');
+    expect(htmlelments.queryAll(By.css('ul')).length).toEqual(2);
+    expect(htmlelments.queryAll(By.css('li')).length).toEqual(mockWihslistProperties.data.property.length);
+  })
+  it('test failure to remove an element in my wishlist', () => {
+    // this was handle by the service
```

Comment:
You can add an assertion here, since its currently empty

Reply:
Done